### PR TITLE
Refactor logging statements in admission controller

### PR DESCRIPTION
**Pull request by omerap12**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
To align the logging with the best practices and guidelines established within the Kubernetes ecosystem.
#### Which issue(s) this PR fixes:

As part of https://github.com/kubernetes/autoscaler/issues/6905, I'm splitting this into multiple PRs rather than submitting a single large PR.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

**Comment by omerap12** [comment]
@raywainman  , fixed the conflicts . PTAL :)

**Review by jackfrancis — APPROVED** [review]
/lgtm

File: vertical-pod-autoscaler/pkg/admission-controller/certs.go [hunk 1]
```diff
@@ -34,11 +34,10 @@ type certsConfig struct {
 func readFile(filePath string) []byte {
 	res, err := os.ReadFile(filePath)
 	if err != nil {
-		klog.Errorf("Error reading certificate file at %s: %v", filePath, err)
+		klog.ErrorS(err, "Error reading certificate file", "file", filePath)
 		return nil
 	}
-
-	klog.V(3).Infof("Successfully read %d bytes from %v", len(res), filePath)
+	klog.V(3).InfoS("Successfully read bytes fron file", "bytes", len(res), "file", filePath)
```
Comment (adrianmoisey) [inline vertical-pod-autoscaler/pkg/admission-controller/certs.go:40]:
```suggestion
	klog.V(3).InfoS("Successfully read bytes from file", "bytes", len(res), "file", filePath)
```

File: vertical-pod-autoscaler/pkg/admission-controller/resource/pod/recommendation/recommendation_provider.go [hunk 2]
```diff
@@ -56,10 +56,10 @@ func GetContainersResources(pod *core.Pod, vpaResourcePolicy *vpa_types.PodResou
 		recommendation := vpa_api_util.GetRecommendationForContainer(container.Name, &podRecommendation)
 		if recommendation == nil {
 			if !addAll {
-				klog.V(2).Infof("no matching recommendation found for container %s, skipping", container.Name)
+				klog.V(2).InfoS("No recommendation found for container, skipping", "container", container.Name)
 				continue
 			}
-			klog.V(2).Infof("no matching recommendation found for container %s, using Pod request", container.Name)
+			klog.V(2).InfoS("No matching found for container, using Pod request", "container", container.Name)
```
Comment (adrianmoisey) [inline vertical-pod-autoscaler/pkg/admission-controller/resource/pod/recommendation/recommendation_provider.go:62]: The wording of "No matching found for container" seems strange, is this better:
```suggestion
			klog.V(2).InfoS("No match found for container, using Pod request", "container", container.Name)
```

File: vertical-pod-autoscaler/pkg/admission-controller/certs.go (hunk 1 above)
Comment (omerap12) [reply vertical-pod-autoscaler/pkg/admission-controller/certs.go:40]: Thanks!

**Comment by adrianmoisey** [comment]
Hey @jbartosik, can you approve this?
/assign jbartosik

**Review by kwiesmueller — APPROVED** [review]
/lgtm
/approve

**Pull request merged**